File: server/internal/server/get_jwks.go
package server

import (
	"context"

	pb "github.com/dritelabs/accounts/internal/proto/v1"
	"google.golang.org/protobuf/types/known/anypb"
)

func (s *AccountServer) GetJwks(ctx context.Context, in *pb.GetJwksRequest) (*anypb.Any, error) {
	return &anypb.Any{
		Value: []byte(s.Jwks),
	}, nil
}

File: frontend/server/old/services2/get-jwks.ts
import { promisify } from "node:util";
import { Metadata } from "@grpc/grpc-js";
import { Any } from "google-protobuf/google/protobuf/any_pb";
import { GetJwksRequest } from "~/server/proto/v1/jwk_pb";
import { client } from "./client";

export async function getJwks() {
  const req = new GetJwksRequest();
  const res = await getJWKS(req, new Metadata());
  const arrayBuff = res.getValue();
  const string = Buffer.from(arrayBuff as Uint8Array).toString("utf8");

  return JSON.parse(string);
}

// const getJWKS = promisify<GetJwksRequest, Metadata, Any>(
//   client.getJwks.bind(client)
// );
